bulk_site_creator/views.py

```diff
@@ -0,0 +1,279 @@
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
+@require_http_methods(["GET", "POST"])
+def new_job(request):
+    sis_account_id = request.LTI["custom_canvas_account_sis_id"]
+    terms, _current_term_id = get_term_data_for_school(sis_account_id)
+    school_id = sis_account_id.split(":")[1]
+    canvas_site_templates = get_canvas_site_templates_for_school(school_id)
+    potential_course_sites_query = None
+    departments = []
+    course_groups = []
+    selected_term_id = None
+    selected_course_group_id = None
+    selected_department_id = None
+
+    # Only display the Course Groups dropdown if the tool is launched in the COLGSAS sub-account
+    if school_id == 'colgsas':
+        try:
+            course_groups = get_course_group_data_for_school(sis_account_id)
+        except Exception:
+            logger.exception(f"Failed to get course groups with sis_account_id {sis_account_id}")
+    # For all other schools, display just the Departments dropdown
+    else:
+        try:
+            departments = get_department_data_for_school(sis_account_id)
+        except Exception:
+            logger.exception(f"Failed to get departments with sis_account_id {sis_account_id}")
+
+    if request.method == "POST":
+        selected_term_id = request.POST.get("courseTerm", None)
+        selected_course_group_id = request.POST.get("courseCourseGroup", None)
+        selected_department_id = request.POST.get("department", None)
+
+        # Retrieve all course instances for the given term_id and account that do not have Canvas course sites
+        # nor are set to be fed into Canvas via the automated feed
+        # TODO Add bulk_processing flag to filter
+        potential_course_sites_query = get_course_instance_query_set(
+            selected_term_id, sis_account_id
+        ).filter(canvas_course_id__isnull=True, sync_to_canvas=0)
+
+    # TODO maybe better to use template tag unless used elsewhere?
+    # TODO cont. this may be included in a summary generation to be displayed in page (see wireframe and Jira ticket)
+    potential_course_site_count = (
+        potential_course_sites_query.count() if potential_course_sites_query else 0
+    )
+
+    context = {
+        "terms": terms,
+        "potential_course_sites": potential_course_sites_query,
+        "potential_site_count": potential_course_site_count,
+        "canvas_site_templates": canvas_site_templates,
+        "departments": departments,
+        "course_groups": course_groups,
+        'selected_term_id': selected_term_id,
+        'selected_course_group_id': selected_course_group_id,
+        'selected_department_id': selected_department_id
+    }
+
+    return render(request, "bulk_site_creator/new_job.html", context=context)
+
+
+@login_required
+@lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
+@require_http_methods(["POST"])
+def create_bulk_job(request: HttpRequest) -> Optional[JsonResponse]:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function calls 18 other functions. By comparison, 98% of the functions in the CodeGuru reference dataset use fewer. This indicates the function is highly coupled with other functions. A function that is highly coupled with other functions is difficult to understand and its behavior might change unexpectedly when one of its referenced functions is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this function or break it into multiple functions.